packages/components/src/higher-order/navigate-regions/index.js

```diff
@@ -37,42 +57,48 @@ export function useNavigateRegions( ref, shortcuts = defaultShortcuts ) {
 		nextRegion.focus();
 		setIsFocusingRegions( true );
 	}
-	const focusPrevious = useCallback( () => focusRegion( -1 ), [] );
-	const focusNext = useCallback( () => focusRegion( 1 ), [] );
 
-	useKeyboardShortcut( shortcuts.previous, focusPrevious, {
-		bindGlobal: true,
-	} );
-	useKeyboardShortcut( shortcuts.next, focusNext, { bindGlobal: true } );
+	const clickRef = useRefEffect(
+		( element ) => {
+			function onClick() {
+				setIsFocusingRegions( false );
+			}
 
-	useEffect( () => {
-		function onClick() {
-			setIsFocusingRegions( false );
-		}
-
-		ref.current.addEventListener( 'click', onClick );
+			element.addEventListener( 'click', onClick );
 
-		return () => {
-			ref.current?.removeEventListener( 'click', onClick );
-		};
-	}, [ setIsFocusingRegions ] );
+			return () => {
+				element.removeEventListener( 'click', onClick );
+			};
+		},
+		[ setIsFocusingRegions ]
+	);
 
-	if ( ! isFocusingRegions ) {
-		return;
-	}
-
-	return 'is-focusing-regions';
+	return {
+		ref: useMergeRefs( [ ref, clickRef ] ),
+		className: isFocusingRegions ? 'is-focusing-regions' : '',
+		onKeyDown( event ) {
+			if (
+				shortcuts.previous.some( ( { modifier, character } ) => {
+					return isKeyboardEvent[ modifier ]( event, character );
+				} )
+			) {
+				focusRegion( -1 );
+			} else if (
+				shortcuts.next.some( ( { modifier, character } ) => {
+					return isKeyboardEvent[ modifier ]( event, character );
+				} )
+			) {
+				focusRegion( 1 );
+			}
+		},
```

Comment:
I think if this becomes the way to write shortcuts, we'll probably need better APIs to make this more friendly. As a shortcut developer, I know three things: the shortcut combination, the element on which to apply it and the callback.

Reply:
I'll think a bit more about how we could simplify connecting this with React events.

Reply:
Yes, we'll most likely need a better API. Currently we have an API to _change_ existing shortcut combinations, but not to add them to a certain region. Based on the section (global/editor, block, text etc.) we can add it to the right element.